marlowe-runtime/test/Language/Marlowe/Runtime/Transaction/ConstraintsSpec.hs

```diff
@@ -936,6 +932,84 @@ genSimpleScriptContext marloweVersion constraints = do
           { payoutOutputs = mempty
           }
 
+shrinkScriptContext
+  :: MarloweVersion v
+  -> TxConstraints BabbageEra v
+  -> Either (MarloweContext v) PayoutContext
+  -> [Either (MarloweContext v) PayoutContext]
```

Comment:
This might be fragile with respect to new capabilities like open roles, plutus oracles, and minting.

Reply:
Yes, these tests are fragile, as is the code under test. It will probably need to be restructured when we need to support new major functionality.